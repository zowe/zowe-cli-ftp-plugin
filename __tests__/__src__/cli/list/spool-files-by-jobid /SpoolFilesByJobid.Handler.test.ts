/*
 * This program and the accompanying materials are made available under the terms of the
 * Eclipse Public License v2.0 which accompanies this distribution, and is available at
 * https://www.eclipse.org/legal/epl-v20.html
 *
 * SPDX-License-Identifier: EPL-2.0
 *
 * Copyright Contributors to the Zowe Project.
 *
 */

import ListSpoolFilesByJobidHandler from "../../../../../src/cli/list/spool-files-by-jobid/SpoolFilesByJobid.Handler";
import TestUtils from "../../TestUtils";

describe("List spool files by job id handler", () => {

    it("should return no spool file if the no spool file is not found.", async () => {
        const handler = new ListSpoolFilesByJobidHandler();
        const files: any[] = [];
        const jobDetails = {
            jobname: "jobname1",
            jobid: "jobid1",
            spoolFiles: files
        };

        const mockResponse = TestUtils.getMockResponse();
        const mockParams: any = {
            arguments: {
                jobid: "jobid1"
            },
            connection: {
                getJobStatus: jest.fn().mockReturnValue(Promise.resolve(jobDetails))
            },
            response: mockResponse
        };
        await handler.processFTP(mockParams);
        expect(mockResponse.data.setMessage.mock.calls[0][0]).toBe("\"0\" spool files obtained for job \"jobname1(jobid1)\"");
        expect(mockResponse.data.setObj.mock.calls[0][0]).toMatchSnapshot();
        expect(mockResponse.format.output.mock.calls[0][0]).toMatchSnapshot();
    });

    it("should return correct message if at least one spool file is found.", async () => {
        const handler = new ListSpoolFilesByJobidHandler();
        const jobDetails = {
            jobname: "jobname1",
            jobid: "jobid1",
            spoolFiles: [
                {
                    name: "file1"
                },
                {
                    name: "file2"
                }
            ]
        };

        const mockResponse = TestUtils.getMockResponse();
        const mockParams: any = {
            arguments: {
                jobid: "jobid1"
            },
            connection: {
                getJobStatus: jest.fn().mockReturnValue(Promise.resolve(jobDetails))
            },
            response: mockResponse
        };
        await handler.processFTP(mockParams);
        expect(mockResponse.data.setMessage.mock.calls[0][0]).toBe("\"2\" spool files obtained for job \"jobname1(jobid1)\"");
        expect(mockResponse.format.output.mock.calls[0][0]).toMatchSnapshot();
        expect(mockResponse.data.setObj.mock.calls[0][0]).toMatchSnapshot();
        expect(mockResponse.format.output.mock.calls[0][0]).toMatchSnapshot();
    });

});
