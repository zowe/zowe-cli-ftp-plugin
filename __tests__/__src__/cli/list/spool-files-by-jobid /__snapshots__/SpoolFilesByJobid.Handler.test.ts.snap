// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List spool files by job id handler should return correct message if at least one spool file is found. 1`] = `
Object {
  "fields": Array [
    "id",
    "ddname",
    "procstep",
    "stepname",
  ],
  "format": "table",
  "output": Array [
    Object {
      "name": "file1",
    },
    Object {
      "name": "file2",
    },
  ],
}
`;

exports[`List spool files by job id handler should return correct message if at least one spool file is found. 2`] = `
Array [
  Object {
    "name": "file1",
  },
  Object {
    "name": "file2",
  },
]
`;

exports[`List spool files by job id handler should return correct message if at least one spool file is found. 3`] = `
Object {
  "fields": Array [
    "id",
    "ddname",
    "procstep",
    "stepname",
  ],
  "format": "table",
  "output": Array [
    Object {
      "name": "file1",
    },
    Object {
      "name": "file2",
    },
  ],
}
`;

exports[`List spool files by job id handler should return no spool file if the no spool file is not found. 1`] = `Array []`;

exports[`List spool files by job id handler should return no spool file if the no spool file is not found. 2`] = `
Object {
  "fields": Array [
    "id",
    "ddname",
    "procstep",
    "stepname",
  ],
  "format": "table",
  "output": Array [],
}
`;
